src/main/java/com/google/cloud/solutions/autotokenize/pipeline/EncryptionPipeline.java

```diff
@@ -153,14 +155,33 @@ public PipelineResult run() throws Exception {
     var encryptedSchema = buildEncryptedSchema();
     var encryptedRecords =
         applyReadAndEncryptionSteps().apply(RecordNester.forSchema(encryptedSchema));
+    String outputType;
 
     if (options.getOutputDirectory() != null) {
-      encryptedRecords.apply(
-          "WriteAVRO",
-          AvroIO.writeGenericRecords(encryptedSchema)
-              .withSuffix(".avro")
-              .to(cleanDirectoryString(options.getOutputDirectory()) + "/data")
-              .withCodec(CodecFactory.snappyCodec()));
+      if (options.getOutputType() != null) {
+         outputType = options.getOutputType();
+      }
+      else{
+        outputType = options.getSourceType().getValueDescriptor().toString();
+      }
+      switch (outputType) {
+        case "AVRO":
+          encryptedRecords.apply(
+                  "WriteAVRO",
+                  AvroIO.writeGenericRecords(encryptedSchema)
+                          .withSuffix(".avro")
+                          .to(cleanDirectoryString(options.getOutputDirectory()) + "/data")
+                          .withCodec(CodecFactory.snappyCodec()));
+          break;
+        case "PARQUET":
+          encryptedRecords.apply(
+                  "WritePARQUET",
+                  FileIO.<GenericRecord>write()
+                          .withSuffix(".parquet")
+                          .via(ParquetIO.sink(buildEncryptedSchema()))
```

Comment:
`buildEncryptedSchema` has already been called, so you can re-use `encryptedSchema`